datafusion/spark/src/function/array/spark_array.rs

```diff
@@ -92,21 +92,6 @@ impl ScalarUDFImpl for SparkArray {
         ))))
     }
 
-    fn return_field_from_args(&self, args: ReturnFieldArgs) -> Result<FieldRef> {
-        let data_types = args
-            .arg_fields
-            .iter()
-            .map(|f| f.data_type())
-            .cloned()
-            .collect::<Vec<_>>();
-        let return_type = self.return_type(&data_types)?;
-        Ok(Arc::new(Field::new(
-            "this_field_name_is_irrelevant",
-            return_type,
-            false,
-        )))
-    }
```

Comment:
I'm actually thinking maybe we keep this and just fold `return_type` logic into here; specifically because it returns `false` for the nullability whereas default implementation for `return_field_from_args` returns `true` for nullability.

- We'd still need to implement `return_type` to satisfy trait, but we can have it return an internal err, see:

https://github.com/apache/datafusion/blob/b52a81db70002489c20bc78dceb33aea2edf44b7/datafusion/functions-nested/src/map_values.rs#L95-L112

Reply:
I'm not familiar with the degree to which DataFusion uses the nullability information of a field, so I can't say what the impact would be. But I assume it's better to be more informative where possible, e.g. we know make array function can't ever return a null so it's good if we can preserve that information, especially since the Spark version already had it.